tools/charts/picker.py

```diff
@@ -0,0 +1,368 @@
+#  H-Lol    |            |     x      |      x     |     x
+#  V-St-Lol |            |            |      x     |     x
+#  H-St-Lol |            |            |      x     |     x
+#
+# Scatter => N-P + 2-V
+#
+
+# Topic Cache JSON
+_TOPIC_CACHE = '../../server/config/nl_page/topic_cache.json'
+_SDG_TOPIC_CACHE = '../../server/config/nl_page/sdg_topic_cache.json'
+
+# Generated charts dir
+_OUT_DIR = 'output_charts'
+
+_HTML_OPEN = """
+<html>
+  <head>
+    <link
+      rel="stylesheet"
+      href="https://dev.datacommons.org/css/datacommons.min.css"
+    />
+    <script src="https://dev.datacommons.org/datacommons.js"></script>
+  </head>
+  <body style="max-width: 40vw; margin: auto">    
+"""
+
+_HTML_CLOSE = """
+  </body>
+</html>
+"""
+
+_MAIN_SPC = '    '
+_SUB_SPC = _MAIN_SPC + '  '
+
+
+def _fname(outdir: str, fno: int) -> str:
+  return os.path.join(outdir, f'shard_{fno}.html')
+
+
+class Writer:
+
+  def __init__(self, outdir):
+    self.next_fno = 1
+    self.ncharts = 0
+    self.total_charts = 0
+    self.fp = None
+    os.makedirs(outdir, exist_ok=True)
+    self.outdir = outdir
+    self._new_html()
+
+  def add(self, chart_type: str, params: List[str]):
+    content = f'{_MAIN_SPC}<datacommons-{chart_type}\n'
+    content += f'{_SUB_SPC}'
+    content += f'\n{_SUB_SPC}'.join(params)
+    content += f'\n{_MAIN_SPC}></datacommons-{chart_type}>\n'
+    self.fp.write(content)
+
+    self.total_charts += 1
+    self.ncharts += 1
+    if self.ncharts >= 100:
+      self._new_html()
+
+  def close(self):
+    if self.fp:
+      self.fp.write(_HTML_CLOSE)
+      self.fp.close()
+
+  def _new_html(self):
+    self.close()
+    self.fp = open(_fname(self.outdir, self.next_fno), 'w')
+    self.fp.write(_HTML_OPEN)
+    # Write the header for html.
+    self.next_fno += 1
+    self.ncharts = 0
+
+
+@dataclass
+class Context:
+  place: str
+  child_type: str
+  child_places: List[str]
+  writer: Writer
+  topic_map: Dict[str, Dict]
+
+
+def _composite(v: str) -> bool:
+  return v.startswith('dc/topic/') or v.startswith('dc/svpg/')
+
+
+#
+# Helper functions to generate the chart configs.
+#
+def _gauge(v: str, title: str, ctx: Context):
+  parts = [
+      f'header="{title}"',
+      f'place="{ctx.place}"',
+      f'variable="{v}"',
+      'min="0"',
+      'max="100"',
+  ]
+  ctx.writer.add('gauge', parts)
+
+
+def _highlight(v: str, title: str, ctx: Context):
+  parts = [f'header="{title}"', f'place="{ctx.place}"', f'variable="{v}"']
+  ctx.writer.add('higlight', parts)
+
+
+def _map(v: str, title: str, ctx: Context):
+  parts = [
+      f'header="{title}"',
+      f'parentPlace="{ctx.place}"',
+      f'childPlaceType="{ctx.child_type}"',
+      f'variable="{v}"',
+  ]
+  ctx.writer.add('map', parts)
+
+
+def _ranking(v: str, title: str, ctx: Context):
+  parts = [
+      f'header="{title}"',
+      f'parentPlace="{ctx.place}"',
+      f'childPlaceType="{ctx.child_type}"',
+      f'variable="{v}"',
+      'rankingCount=10',
```

Comment:
`rankingCount` isn't supported by the web components yet so this line and line 171 will get ignored.

Reply:
Update: working on adding this now, we can leave this in for now!